### ISSUE TITLE
How to merge latest development branch into existing fork?

### ISSUE THREAD

**Issue by cckweiss** [AKSW/OntoWiki#317]
Let me begin by stating that I am NOT a developer. I'm a systems administrator. I was asked to move an application called UDFR (http://udfr.org) from an old Novell SLES system to an Amazon Linux AWS instance.

I was not able to simply make a tarball of the old code and move it. It would not run. There were too many dependencies on deprecated libraries that were not available on Amazon Linux.

I rebuilt all the component elements (Virtuoso, Apache httpd, PHP, and OntoWiki) on Amazon Linux, and was able to get the default sample application in Ontowiki to display (http://udfr-tmp.cdlib.org/ontowiki/). I have all the infrastructure working. My challenge now is to get the UDFR application moved across.

The UDFR application is a fork of the AKSW OntoWiki development branch (https://github.com/UDFR/Ontowiki.git). The changes and additions that were made to the base OntoWiki software are not neatly segregated - lots and lots of files were edited, all over the place, to create the UDFR application.

Is there a way to merge all of the changes that were made to the main AKSW development branch into the UDFR fork, without destroying all of the changes that were made in the UDFR fork? I want to end up with your current development branch, which works just perfectly for me, plus all the changes that were made by the UDFR developers.

I assume there has to be a way to replay the UDFR edits and additions into the current AKSW OntoWiki development branch, but I don't know how. Can you help me out?

--Ken

**Comment by white-gecko** [comment]
There is quite a big diff between your branch an our current master and develop branch. So I think it will need quite some hand crafting to merge your changes into our main branch.
We would really appreciate if you could contribute your fixes and features in pull-requests to our develop branch.

For adding customizations to OntoWiki while keeping it upstream compatible is best to create custom extensions, while contributing fixes as pull-requests.

**Comment by cckweiss** [comment]
Natanael,

Yes, after looking at the number of conflicts when I attempted to merge the
codelines, and digging into the change history for our branch of your code,
I concluded that there was no good way to complete a merge.

Unfortunately, there are two factors that make it impossible to contribute
our features as pull-requests. First, our code is abandon-ware. There has
been no active development for over 4 years, and there is no one on staff
now with any knowledge of the code at all. Second, the features were not
architected as custom extensions. Instead, there are hundreds of changes to
many of the core .php files throughout the OntoWiki code. It's just a mess,
a giant hairball. There is no way to cleanly separate the original OntoWiki
code from the changes that were made to support our application.

We've decided to continue to run our branch as-is. I was able to get it to
run on current versions of Apache, PHP, and Virtuoso, so at least all the
infrastructure is up-to-date.

Thanks for responding, and I'm sorry I can't work more closely with your
team.


--Ken

cckweiss@gmail.com
